internal/kubernetes/operator/dbusers/dbusers_test.go

```diff
@@ -0,0 +1,356 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbusers
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/secrets"
+	atlasV1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"go.mongodb.org/atlas/mongodbatlas"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_convertUserLabels(t *testing.T) {
+	t.Run("Can convert user labels from Atlas to the Operator format", func(t *testing.T) {
+		atlasUser := &mongodbatlas.DatabaseUser{
+			Labels: []mongodbatlas.Label{
+				{
+					Key:   "TestKey",
+					Value: "TestValue",
+				},
+			},
+		}
+
+		expectedLabels := []common.LabelSpec{
+			{
+				Key:   "TestKey",
+				Value: "TestValue",
+			},
+		}
+
+		if got := convertUserLabels(atlasUser); !reflect.DeepEqual(got, expectedLabels) {
```

Comment:
[not-blocking] two things you can use instead of reflect, `github.com/go-test/deep` or `github.com/stretchr/testify`

example of deep
https://github.com/mongodb/mongodb-atlas-cli/blob/22bbf41abb2d5c7476294273000a807301912450/internal/cli/atlas/clusters/clusters_test.go#L254

Example of testify
https://github.com/mongodb/mongodb-atlas-cli/blob/3d0bea5f6b288cc3b4dc7447748f7a0fa8c59c21/internal/cli/atlas/backup/schedule/update_test.go#L112

Both pick at random

Reply:
I'd prefer to leave an explicit version, but I can be talked into one of those you offered, if this is a problem with tests consistency.